quadcontrol/control.c

```diff
@@ -303,6 +388,42 @@ static int quad_run(void)
 }
 
 
+static void quad_rcbusHandler(const rcbus_msg_t *msg)
+{
+	if (msg->channelsCnt < RC_CHANNELS_CNT) {
+		fprintf(stderr, "quad-control: rcbus supports insufficient number of channels\n");
+		return;
+	}
+
+	/* Manual Arm: SWA == MAX, SWB == MIN and Throttle == 0 and scenario cannot be launched */
+	if (msg->channels[RC_SWA_CH] >= (0.95 * MAX_CHANNEL_VALUE) && msg->channels[RC_SWB_CH] <= (MIN_CHANNEL_VALUE + 0.05 * MAX_CHANNEL_VALUE)
```

Comment:
This will cause uint16_t -> double -> uint16_t series of casts. How about:
```suggestion
	if (msg->channels[RC_SWA_CH] >= ((95 * MAX_CHANNEL_VALUE) / 100) && msg->channels[RC_SWB_CH] <= (MIN_CHANNEL_VALUE + (5 * (MAX_CHANNEL_VALUE - MIN_CHANNEL_VALUE))) / 100
```
Also we should imho take 5% of MAX - MIN, not MAX alone, as we want to get 5% of the dynamic range, not absolute max. Rationale - it wouldn't work very well if e.g. MIN = 1000 MAX = 1100.

Perhaps new defines for trigger values, so this conditions won't get so complicated?

Reply:
Changed as suggested.